idl/github.com/uber/cadence/shared.thrift

```diff
@@ -805,3 +810,26 @@ struct ListClosedWorkflowExecutionsResponse {
   10: optional list<WorkflowExecutionInfo> executions
   20: optional binary nextPageToken
 }
+
+struct QueryWorkflowRequest {
+  10: optional string domain
+  20: optional WorkflowExecution execution
+  30: optional TaskList taskList
```

Comment:
Can we figure our the tasklist automatically? the start workflow record of the workflow ID should have it right?

Reply:
TaskList will be optional for running workflow which we could load taskList from mutable state. For closed workflow execution, if the mutable state is removed, we won't be able to get the task list.

Reply:
We will remove taskList from query request, and fetch it from history. This will make client easier.